### DISCUSSION TITLE
Make SnippetViewSet Compatible with Django SimpleListFilter

### DISCUSSION THREAD

**Discussion by rjbullock** [wagtail/wagtail#12142] [Ideas]
We use SimoleListFilter extensively. It was compatible with ModelAdmin but is not with SnippetViewSet. Can we bring back that functionality?

**Comment by gasman** [comment]
Unlikely, sorry. `SimpleListFilter` is designed for use with the Django admin module, and interoperating with it required ModelAdmin to tap into the undocumented internals of `django.contrib.admin` - which was one of the driving factors for moving away from ModelAdmin.

The new snippets interface uses [django-filter](https://django-filter.readthedocs.io/en/main/) for filtering, and while that isn't compatible with the Django admin's filter modules, it should provide the same functionality and more.
